proposals/0002-root-signature-in-clang.md

```diff
@@ -936,6 +936,199 @@ Operands:
+Usage:
+
+```
+  -force-rootsig-ver rootsig_1_0
+  -force-rootsig-ver rootsig_1_1
+```
+
+Behaviour:
+
+ - Validation logic that is specific to the version will be followed, as
+described above
+ - Serialized root signature in DXIL container will be compliant with specified
+version
+
+Design Notes: Required for backwards compatibility. Implemented in Clang.
+
+#### Option `-rootsig-define`:
+
+Sets the root signature attribute for the entry function to be the root
+signature defined by the given macro expansion.
+
+Usage:
+
+Given a defined macro either provided in the source file
+(`#define RS "CBV(b0)"`), or, as a command line define
+(`-D RS="CBV(b0)"`)
+
+```
+  -rootsig-define RS
+```
+
+Behaviour:
+
+ - If the entry function does not have a root signature attribute, it will use
+the one defined by the macro expansion
+
+ - If the entry function has a root signature attribute, it will overwrite to
+use the one defined by the macro expansion
+
+Design Notes: Commonly used. Planned in Clang.
+
+#### Target Root Signature Version
+
+Compiles the "entry" root signature, specified by the given macro expansion, to
+a DXIL Container with just the (version specific) RTS0 part.
+
+Usage:
+
+Given a defined macro in the source file (`#define RS "CBV(b0)"`)
+
+```
+  -T <root signature version> -E <entry root signature>`
+  -T rootsig_1_0 -E RS /Fo RS.bin
+  -T rootsig_1_1 -E RS /Fo RS.bin
+```
+
+Behaviour:
+
+ - Parse and perform syntactic validations of "entry" root signature
+ - Perform the 'basic' sub-set of validations that are available without access
+to resource bindings
+ - Produces a DXIL container with only the serialized "entry" root signature
+as the only part
+
+ - Note: It is not possible to use `-rootsig-define` to overwrite which root
+signature will be used as the "entry" root signature.
+
+Design Notes: Commonly used. Planned in Clang.
+
+#### Option `-Qstrip-rootsignature`
+
+Omits the root signature part (RTS0) from the produced DXIL Container.
+
+Usage:
+
+```
+  -Qstrip-rootsignature
+```
+
+Behaviour:
+
+ - Parse and perform syntactic validations of "entry" root signature
+ - Perform the 'basic' sub-set of validations that are available without access
+to resource bindings
+ - Produces the DXIL container with the RTS0 omitted
+
+Design Notes: Usable within .hlsl source files. Planned in Clang unless there is
+a signal of it being unused.
+
+Design Notes: Plan to match DXC behaviour directly such that we still do
+the parsing and basic validations of the root signataure.
+
+#### Option `-setrootsignature`
+
+Overwrite the root signature part (RTS0) with the RTS0 of provided DXIL
+Container.
+
+Usage:
+
+```
+  -setrootsignature <DXIL Container>
+```
+
+Behaviour:
+
+ - Insert/overwrite the RTS0 part of the compiled DXIL Container with the RST0
+contents of the input container
+ - Does not perform any validation with RTS0 of the provided DXIL Container
+ - Requires input to be a valid DXIL container with or without an RST0 part
```

Comment:
Which input?  The input container for the `-setrootsignature` option is required to be a valid DXIL container *with* an `RTS0` part.

The shader input (whether compiling from HLSL or loading a DXIL Container) need not have a root signature attached.

If one is present in the shader input, that root signature must still be well-formed, but it will not be checked against the shader for binding compatibility, and it will not be used for the final DXIL Container output.